Funcky.Test/Extensions/EnumerableExtensions/WithFirstTest.cs

```diff
@@ -0,0 +1,43 @@
+using System.Linq;
+using Funcky.Extensions;
+using Xunit;
+
+namespace Funcky.Test.Extensions.EnumerableExtensions
+{
+    public class WithFirstTest
+    {
+        [Fact]
+        public void AnEmptySequenceWithFirstReturnsAnEmptySequence()
+        {
+            var emptySequence = Enumerable.Empty<string>();
+
+            Assert.Empty(emptySequence.WithFirst());
+        }
+
+        [Fact]
+        public void ASequenceWithOneElementWithFirstHasOneElementWhichIsMarkedFirst()
+        {
+            const string expectedValue = "Hello world!";
+            var oneElementSequence = Enumerable.Repeat(expectedValue, 1);
+
+            foreach (var (value, isFirst) in oneElementSequence.WithFirst())
```

Comment:
This test wouldn't notice if the sequence were empty

Reply:
Adressed in 11308f40